recipes/vvenc/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+
+
+class vvencRecipe(ConanFile):
+    name = "vvenc"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Fraunhofer Versatile Video Encoder (VVenC)"
+    topics = ("video", "encoder", "codec", "vvc", "h266")
+    homepage = "https://www.hhi.fraunhofer.de/en/departments/vca/technologies-and-solutions/h266-vvc.html"
+    package_type = "library"
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    def validate_build(self):
+        if conan_version.major == 2:
+            self._validate_build2()
+        elif conan_version.major == 1:
+            self._validate_build1()
+
+    def _validate_build1(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, 14)
+        # compiler.cppstd isn't set! but we still continue here in V1,
+        # imagining just like compiler.cppstd was set to 14
+        # while package_id doesn't reflect that at all, and cppstd_default also
+        # might be different from 14, we silently force -std=c++14 to be
+        # specified during the build. it may produce more incompatibilities,
+        # and break user's expectation
+        # (like, output binary depends really on C++14 symbols libstdc++.so)
+        # it's V1 design flaw which isn't going to be addressed here
+        # (and probably nowhere, because conan V1 is going to be discontinued in CCI)
+        # once V1 is retired, that code will be removed altogether
+        self.output.warning("compiler.cppstd is not set, but we assume C++14")
+
+    def _validate_build2(self):
+        # validates the minimum and maximum C++ standard supported
+        # currently, the project can only be built with C++14 standard
+        # it cannot be built with older standard because
+        # it doesn't have all the C++ features needed
+        # and it cannot be built with newer C++ standard
+        # because they have existing C++ features removed
+        check_min_cppstd(self, 14)
+        # FIXME: linter complains, but function is there
+        # https://docs.conan.io/2.0/reference/tools/build.html?highlight=check_min_cppstd#conan-tools-build-check-max-cppstd
+        check_max_cppstd = getattr(sys.modules['conan.tools.build'], 'check_max_cppstd')
+        check_max_cppstd(self, 14)
+
+    def package_id(self):
+        # still important, older binutils cannot recognize
+        # object files created with newer binutils,
+        # thus linker cannot find any valid object and therefore symbols
+        # (fails to find `vvenc_get_version`, which is obviously always there)
+        # this is not exactly modeled by conan right now, 
+        # so "compiler" setting is closest thing to avoid an issue
+        # (while technically it's not a compiler, but linker and archiver)
+        # del self.info.settings.compiler
+        pass
```

Comment:
what is the purpose of defining the method like this? why not remove the settings? We need more clarification around this

Reply:
I don't know what else to clarify here... I tried to explain as detailed as possible.
if you remove `del self.info.settings.compiler` then GCC 5 and GCC 10 (example only) will try to use the same binary.
if it happened to be built via GCC 10 first, then object files (.o) will be created by newer binutils.
afterwards, GCC 5 build will see package is already there (upstream available) and will just run to execute the test_package.
it will execute older linker (`ld` from binutils), but linker doesn't know newer elf format and doesn't see any symbols. this causes missing symbol error (unresolved reference).
it actually happened in CI (see [bot comment](https://github.com/conan-io/conan-center-index/pull/17025#issuecomment-1618810701)) and reproduced locally for me.

P.S. let's try to avoid questions like `why not X` and `why not Y`. ofcourse, there is always a room for improvement and you cannot reach 100% ideal of the perfection. but I think it's already good for the first iteration as it passes v1 and v2 CI workflows. all the further improvements could be done in sequential PRs by demand.

Reply:
Thanks a lot for the clarification, now I can understand the issue, this is the reason why I was asking for clarification, not to really improve the recipe further